sda/cmd/s3inbox/proxy.go

```diff
@@ -199,6 +200,19 @@ func (p *Proxy) allowedResponse(w http.ResponseWriter, r *http.Request, token jw
 			return
 		}
 
+		// The following block is for treating the case when the client loses connection to the server and then it reconnects to a
+		// different instance of s3inbox. For more details see #1358.
+		if p.fileIds[r.URL.Path] == "" {
+			p.fileIds[r.URL.Path], err = p.database.GetFileIDByUserPathAndStatus(username, filepath, "registered")
+			if err != nil {
+				p.internalServerError(w, r, fmt.Sprintf("failed to retrieve fileID from database: %v", err))
```

Comment:
Was thinking on how smart it is to return all base errors to the client but it is done in so many places that we need to do it all at once.

Reply:
Indeed, we shouldn't propagate the error msg to the client. It should be enough to report the errorCode.  This can change centrally by deprecating [this](https://github.com/neicnordic/sensitive-data-archive/blob/8d0998f20b55035cd166543a6707a77288cf54fc/sda/cmd/s3inbox/proxy.go#L538). I see now that the `log(err)` is done in two places as well.

But better leave this for a new issue so that the team can discuss on it. I opened #1380 about it.